### ISSUE TITLE
opencv can mix qt5/qt4

### ISSUE THREAD

**Issue by ReubenM** [slowmoVideo/slowmoVideo#41]
I have a system with both qt4 and qt5 installed, and I've spent the last hour pulling what remains of my hair out trying to figure out why the build wasn't working. Actually it works just fine. It builds without any problems and immediately segfaults when starting it up. Gdb revealed it was segfaulting in a qt4 library. Which was odd because I had checked and dug through every build file trying to figure out how the heck the executables were getting linked against the qt4 libraries.

Well, turns out that if opencv is built with qt4 support enabled, it pulls in the qt4 libraries with it, which sabotage the build. Rebuilding without qt4 support in opencv fixes everything.

So, is there perhaps a means to move libs pulled in by opencv to the back of the line so they don't make false positive symbol matches that are supposed to be provided by qt5?

**Comment by pmo-19** [comment]
I am using Ubuntu Trusty 14.04. Since Trusty both Qt4 and Qt5 are part of the system. 

By default OpenCV CMakeLists.txt file set QT support as OFF:
`OCV_OPTION(WITH_QT             "Build with Qt Backend support"               OFF  IF (NOT ANDROID AND NOT IOS) )`

There is no Qt4 or Qt5 specific switch. If you set this switch to ON then it will check first for the availability of Qt5 then Qt4, if none is found you end up with an error.

By default the opencv build by Debian / Ubuntu does not support Qt. So better to stay the course.

When building `slowmoVideo` you have a `cmake` switch as well, allowing you to force the use of Qt4 if Qt5 is detected on your system (by default if Qt5 is found, `slowmoVideo` will build wth Qt5).

I use the `cmake` option `-DFORCE_QT4=ON` to build on Qt4.

So don't see your point of modifying the project source. To me it's working fine.

**Comment by ReubenM** [comment]
It's not about modifying the source, it's about checking that opencv is using the same version of qt. The build system pulls in the libraries that opencv are linked to, including qt4/5. It doesn't matter if you configure cmake to use qt5 if opencv is using qt4. In that scenario it ends up linking qt5 code to qt4 libraries. That will successfully build since the API interfaces are similar, but of course will not run.

**Comment by pmo-19** [comment]
Well, I believe its only your business to see how you configure the build of other packages in your system. Here OpenCV.

However, as I said before I don't see the need to build OpenCV with Qt support. OpenCV is a low level library and is ill fit to manage end user Qt applications. As a reference I gave you above, none of the distributions I know of (Debian and Ubuntu) does build OpenCV with Qt Support.

**Comment by valgit** [comment]
@pmo-19  I agree with you, and don't think that compiling slowmovideo need an opencv 'qt aware".
